snapshot/git/gitfiler/pool.go

```diff
@@ -80,6 +92,7 @@ func (p *RepoPool) loop() {
 			p.freeList = append(p.freeList, r)
```

Comment:
This is strange wrt size vs freeList - size is intended to indicate "allocated" repos, vs "reserved" repos.  The double use of "Get()" is confusing here also.  In the cloner case we first Get() the clone, which is more of an expensive allocation operation. Then the user of the pool Get()s the repo, which is more like saying "hey I'm going to use that one."  So I this introduces a repo allocation vs reservation/use issue.

To be clear, size should only increase when: we initialize the pool with existing repos, or in the pool loop we run out of free repos, have spare capacity, and call RepoGetter.Get().